### PR TITLE
Refactor publish dialog step contents.

### PR DESCRIPTION
**Pull request by ILMTitan**
Split the publish dialog step contents into smaller individual control.
Move sub-controls that exist on multiple steps into common controls.
Add Validation.ErrorTemplate to the base ComboBox style.

### CODE REVIEW

File: GoogleCloudExtension/GoogleCloudExtension.Utils/NullEmptyConverter.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            if (value is IConvertible convertible)
+            {
+                stringValue = convertible.ToString(culture);
+            }
+            else
+            {
+                stringValue = value?.ToString();
+            }
+
+            return string.IsNullOrWhiteSpace(stringValue) ^ Invert ? EmptyValue : NotEmptyValue;
```
Comment (redborian) [inline GoogleCloudExtension/GoogleCloudExtension.Utils/NullEmptyConverter.cs]: Although this is neat to write, it is not intuitive. 
`if (Invert) isEmpty = !isEmpty;` would be easier to understand.

File: GoogleCloudExtension/GoogleCloudExtension/PublishDialog/Steps/Gke/GkeInputsControl.xaml [hunk 2]
```diff
@@ -0,0 +1,87 @@
+﻿<UserControl x:Class="GoogleCloudExtension.PublishDialog.Steps.Gke.GkeInputsControl"
```
Comment (redborian) [inline GoogleCloudExtension/GoogleCloudExtension/PublishDialog/Steps/Gke/GkeInputsControl.xaml]: Is there a reason why we don't add licence headers to xaml files?

File: GoogleCloudExtension/GoogleCloudExtension.Utils.UnitTests/GoogleCloudExtension.Utils.UnitTests.csproj [hunk 3]
```diff
@@ -64,10 +64,13 @@
       <HintPath>..\packages\System.ValueTuple.4.4.0\lib\netstandard1.0\System.ValueTuple.dll</HintPath>
     </Reference>
     <Reference Include="System.Xaml" />
+    <Reference Include="WindowsBase" />
```
Comment (redborian) [inline GoogleCloudExtension/GoogleCloudExtension.Utils.UnitTests/GoogleCloudExtension.Utils.UnitTests.csproj]: Why is this needed in UnitTest project?

Comment (ILMTitan) [reply GoogleCloudExtension/GoogleCloudExtension.Utils.UnitTests/GoogleCloudExtension.Utils.UnitTests.csproj]: This was a relic of a test I got rid of. Removing.

File: GoogleCloudExtension/GoogleCloudExtension/PublishDialog/Steps/Gke/GkeInputsControl.xaml (hunk 2 above)
Comment (ILMTitan) [reply GoogleCloudExtension/GoogleCloudExtension/PublishDialog/Steps/Gke/GkeInputsControl.xaml]: There is no good reason, fixing.

File: GoogleCloudExtension/GoogleCloudExtension.Utils/NullEmptyConverter.cs (hunk 1 above)
Comment (ILMTitan) [reply GoogleCloudExtension/GoogleCloudExtension.Utils/NullEmptyConverter.cs]: Done.

**Comment by ILMTitan** [comment]
@redborian PTAL

File: GoogleCloudExtension/GoogleCloudExtension.Utils.UnitTests/NullEmptyBoolConverterTests.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        private NullEmptyBoolConverter _objectUnderTest;
+
+        [TestInitialize]
+        public void BeforeEach()
+        {
+            _objectUnderTest = new NullEmptyBoolConverter();
+        }
+
+        [TestMethod]
+        public void TestConvert_EmptyReturnsTrue()
```
Comment (quoctruong) [inline GoogleCloudExtension/GoogleCloudExtension.Utils.UnitTests/NullEmptyBoolConverterTests.cs]: Maybe we should add tests for converting null and whitespace too?

File: GoogleCloudExtension/GoogleCloudExtension.Utils.UnitTests/NullEmptyConverterTests.cs [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+        [TestMethod]
+        public void TestConvertEmpty()
+        {
+            object val = _objectUnderTest.Convert("", null, null, null);
+
+            Assert.AreEqual(s_emptyResult, val);
+        }
+
+        [TestMethod]
+        public void TestConvertWhitespace()
```
Comment (quoctruong) [inline GoogleCloudExtension/GoogleCloudExtension.Utils.UnitTests/NullEmptyConverterTests.cs]: We can also use `DataRow` to combine the `null`, `""` and `WhiteSpaceString` tests here.

File: GoogleCloudExtension/GoogleCloudExtension.Utils/NullEmptyConverter.cs [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture) =>
+            throw new NotSupportedException();
+
+        /// <summary>
+        /// Implement interface MarkupExtension.
+        /// </summary>
+        public override object ProvideValue(IServiceProvider serviceProvider) => this;
+    }
+}
```
Comment (quoctruong) [inline GoogleCloudExtension/GoogleCloudExtension.Utils/NullEmptyConverter.cs]: nit: new line.

File: GoogleCloudExtension/GoogleCloudExtension.Utils.UnitTests/NullEmptyBoolConverterTests.cs (hunk 4 above)
Comment (ILMTitan) [reply GoogleCloudExtension/GoogleCloudExtension.Utils.UnitTests/NullEmptyBoolConverterTests.cs]: I left that out because that logic is in the base class, and the base class tests are where we test that.
Updating the test names to reflect that.

File: GoogleCloudExtension/GoogleCloudExtension.Utils.UnitTests/NullEmptyConverterTests.cs (hunk 5 above)
Comment (ILMTitan) [reply GoogleCloudExtension/GoogleCloudExtension.Utils.UnitTests/NullEmptyConverterTests.cs]: Done.

**Comment by ILMTitan** [comment]
PTAL
